openapi.yaml

```
@@ -2973,12 +2973,20 @@ paths:
       description: >-
         Calculates an estimate for a batch job. Back-ends can decide to
         either calculate the duration, the costs, the size or a combination of
-        them. This MUST be the upper limit of the incurring costs. Clients can
-        be charged less than specified, but never more. Back-end providers MAY
-        specify an expiry time for the estimate. Starting to process data
-        afterwards MAY be charged at a higher cost. Costs MAY NOT include
-        downloading costs. This can be indicated with the `downloads_included`
-        flag.
+        them.
+
+        Back-end providers MAY specify an expiry time for the estimate.
+        Starting to process data afterwards MAY be charged at a higher cost.
+        Costs MAY NOT include downloading costs. This can be indicated with the
+        `downloads_included` flag.
+        
+        The estimate SHOULD be the upper limit of the costs, but back-end are
+        free to use the field according to their terms of service.
+        
+        For some batch jobs it is not (easily) possible to estimate the costs
+        reliably, e.g. if a user-submitted UDF is part of the process.
```

"user-submitted UDF" sounds like a pleonasm

---

I had to google "pleonasm" and I don't think I agree. I was thinking that in the future there could also be back-end provided UDFs which they might be able to give estimates for, although the question then is why it's not a pre-defined process, indeed. But I'd also be fine to remove the user-submitted here.

```suggestion
        reliably, e.g. if a UDF is part of the process.
```

---

ok I understand.

Another example to consider here: doing ML prediction with a ML model that is "user-submitted" in the same sense (black box)